package/tests/test_server.py

```diff
@@ -243,6 +261,128 @@ def test_nodes_endpoint(cli_runner, client):
+    ):  # pylint: disable=unused-argument
+        return {"context": mock.Mock()}[key]
+
+    mock.patch("kedro_viz.server.get_project_context", new=get_project_context).start()
+    result = cli_runner.invoke(server.commands, ["viz", "--pipeline", "another"])
+    assert "`--pipeline` flag was provided" in result.output
+
+
+@mock.patch("kedro.__version__", "0.15.0")
+def test_viz_kedro15_invalid(cli_runner):
+    """Test that running viz in Kedro 0.15.0,
+    and it is outside of a Kedro project root."""
+
+    def get_project_context(key: str, **kwargs):
+        raise KedroContextError
+
+    mock.patch("kedro_viz.server.get_project_context", new=get_project_context).start()
+    result = cli_runner.invoke(server.commands, "viz")
+    assert "Could not find a Kedro project root." in result.output
+
+
+@mock.patch("kedro.__version__", "0.14.0")
+def test_viz_kedro14(cli_runner):
+    """Test that running viz in Kedro 0.14.0."""
+
+    def create_catalog(config):  # pylint: disable=unused-argument,bad-continuation
+        return lambda x: None
+
+    def get_config(
+        project_path: str, env: str = None  # pylint: disable=bad-continuation
+    ):  # pylint: disable=unused-argument
+        return "config"
+
+    def get_project_context(key: str):
+        return {
+            "create_pipeline": create_pipeline,
+            "create_catalog": create_catalog,
+            "get_config": get_config,
+        }[key]
+
+    mock.patch("kedro_viz.server.get_project_context", new=get_project_context).start()
+    result = cli_runner.invoke(server.commands, "viz")
+    assert result.exit_code == 0, result.output
+
+
+@mock.patch("kedro.__version__", "0.14.0")
+def test_viz_kedro14_pipeline_flag(cli_runner):
+    """Test that running viz with `--pipeline` flag in Kedro 0.14.0 is not supported."""
+
+    def create_catalog(config):  # pylint: disable=unused-argument,bad-continuation
+        return lambda x: None
+
+    def get_config(
+        project_path: str, env: str = None  # pylint: disable=bad-continuation
+    ):  # pylint: disable=unused-argument
+        return "config"
+
+    def get_project_context(key: str):
+        return {
+            "create_pipeline": create_pipeline,
+            "create_catalog": create_catalog,
+            "get_config": get_config,
+        }[key]
+
+    mock.patch("kedro_viz.server.get_project_context", new=get_project_context).start()
+    result = cli_runner.invoke(server.commands, ["viz", "--pipeline", "another"])
+    assert "`--pipeline` flag was provided" in result.output
+
+
+@mock.patch("kedro.__version__", "0.14.0")
+def test_viz_kedro14_invalid(cli_runner):
+    """Test that running viz in Kedro 0.14.0,
+    while outside of a Kedro project is not supported."""
+
+    def get_project_context(key: str, **kwargs):
+        raise KeyError
+
+    mock.patch("kedro_viz.server.get_project_context", new=get_project_context).start()
```

Comment:
```suggestion
    mock.patch("kedro_viz.server.get_project_context", side_effect=KeyError).start()
```

Reply:
Nice, I didn't know about this. Thanks :)